pkg/apis/core/v1beta1/defaults_controllerregistration.go

```diff
@@ -19,7 +20,15 @@ func SetDefaults_ControllerResource(obj *ControllerResource) {
 
 	if obj.Kind == "Extension" {
 		if obj.GloballyEnabled == nil {
-			obj.GloballyEnabled = ptr.To(false)
+			if slices.Contains(obj.AutoEnable, AutoEnableModeShoot) {
+				obj.GloballyEnabled = ptr.To(true)
+			} else {
+				obj.GloballyEnabled = ptr.To(false)
+			}
+		}
+
+		if len(obj.AutoEnable) == 0 && *obj.GloballyEnabled {
+			obj.AutoEnable = []AutoEnableMode{AutoEnableModeShoot}
```

Comment:
Don't we also have to build conversion? If someone sets `globallyEnabled=false`, the `shoot` mode should disappear from `autoEnable[]` (and vice versa)?

Reply:
I can add this 👍

Reply:
Added in https://github.com/gardener/gardener/pull/11982/commits/89af945070e59f10d558af99b6ad457183128f1a